### ISSUE TITLE
Working with default values

### ISSUE THREAD

**Issue by 7even** [lotus/model#155]
Currently `lotus-model` doesn't respect default values in the database:

``` sql
CREATE TABLE deployments (
    id integer NOT NULL,
    project_id integer NOT NULL,
    environment character varying NOT NULL,
    branch character varying NOT NULL,
    deployed_at timestamp with time zone DEFAULT now() NOT NULL
);
```

``` ruby
class Deployment
  include Lotus::Entity
  attributes :id, :project_id, :environment, :branch, :deployed_at
  
  attr_accessor :project
end

deployment = Deployment.new(project_id: 1, environment: 'production', branch: 'master')
DeploymentRepository.persist(deployment)
# Sequel::NotNullConstraintViolation: PG::NotNullViolation: ERROR:  null value in column "deployed_at" violates not-null constraint
# DETAIL:  Failing row contains (4, 1, production, master, null).
```

Seems like the empty `deployed_at` value is being written to the database. I see 2 possible ways to solve this issue: to read the default value into the entity upon it's creation (but I believe it breaks incapsulation and is not very helpful with a default value of `NOW()`), or to save only attributes that were changed.

**Comment by jodosha** [comment]
@7even Thanks for reporting this. Which version are you using?

**Comment by 7even** [comment]
The latest one, `lotus-model (0.2.4)`.

**Comment by jodosha** [comment]
@7even Can you please also paste here the mapping for that entity?

**Comment by 7even** [comment]
Yep, here it is:

``` ruby
collection :deployments do
  entity     Deployment
  repository DeploymentRepository
  
  attribute :id,          Integer
  attribute :project_id,  Integer
  attribute :environment, String
  attribute :branch,      String
  attribute :deployed_at, Time
end
```

**Comment by luishurtado** [comment]
Although there is something to fix regarding the NOT NULL issue, indicating that a given attribute is required and it has a default value should also be clearly defined in the application level and not rely in the database.

**Comment by joneslee85** [comment]
@7even this has been addressed in #161
